client/src/pages/dashboard/index.jsx

```diff
@@ -71,15 +79,19 @@ export default function Page() {
               <h1 className="text-4xl font-bold text-gray-900">{dashboardMetricsFetchErrorDetails.statusCode}: {dashboardMetricsFetchErrorDetails.errorMessage}</h1>
               <p className="mt-4 text-lg text-gray-600">{dashboardMetricsFetchErrorDetails.errorDescription}</p>
               <button 
-                  onClick={() => router.push('/dashboard')} 
+                  onClick={() => router.push('/signin')} 
                   className="mt-8 px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 focus:outline-none focus:bg-blue-600"
               >
-                  Refresh
+                  Sign In
```

Comment:
Having this in subsequent calls when the token expires is a good principle. But I don't think the same applied when we first make the call. When I first launch /dashboard this is what I get:

<img width="1503" alt="Screen Shot 2024-02-29 at 12 59 00 PM" src="https://github.com/rohithaug/UCSD-CSE-210-Team-12/assets/64318022/db246a76-f3b6-4622-816d-cc800a787e39">

This doesn't seem to be the best way, if not logged in then we should directly redirect to the sign in page rather than showing a 401 Unauthorized error.

- Having this for subsequent calls is fine, but for the first time it should directly take to the login page.
- Or you could always take to the login page

Reply:
Ah my bad, I just placed the piece of code to check if `token` exists after the `fetchDashboardMetricsError` instead of above it.